datafusion/optimizer/src/unwrap_cast_in_comparison.rs

```diff
@@ -639,12 +693,12 @@ mod tests {
         // but the type of c6 is uint32
         // the rewriter will not throw error and just return the original expr
         let schema = expr_test_schema();
-        let expr_input = cast(col("c6"), DataType::Int64).eq(lit(0i64));
+        let expr_input = cast(col("c6"), DataType::Float64).eq(lit(0f64));
```

Comment:
the comment of `"c6 > 0" will be cast to `cast(c6 as int64) > 0` should be change to `"c6 > 0" will be cast to `cast(c6 as float64) > 0`

Reply:
Fixed in c6946f065